doc/compiler/runtime/CodeMetaData.md

```diff
@@ -0,0 +1,88 @@
+!--
+Copyright (c) 2016, 2017 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+-->
+
+# Code Metadata and Code Metadata Manager in Testarossa
+
+## Introduction
+
+This document explains and documents the concepts and implementation of runtime
+code metadata and code metadata manager in Testarossa.
+
+## Code Metadata
+
+Code metadata contains runtime metadata information about a compilation unit. 
+Metadata may include start address of allocated code memory and its size, 
+start PC address of compiled code (when it is invoked from an interpreter 
+or invoked from another piece of compiled code), end PC address of compiled code, 
```

Comment:
should this be end PC or more correctly the address at which the method body ends since end PC could be misinterpreted - we could return somewhere part way through the generated code and so the end PC would not match this description. Maybe I'm just reading this differently to you?

Reply:
Because in the code we refer to the end of the method body as end PC (or something similar), I think it should be left as "end PC" but that term could be further clarified.

Reply:
Ok I'm good with that idea - consistency with the code is good, but I agree a clarification is probably a good idea to prevent the slight confusion I had with the term.